Mathlib/Data/Matroid/IndepAxioms.lean

```diff
@@ -167,23 +160,19 @@ namespace IndepMatroid
   (indep_subset := indep_subset)
   (indep_aug := by
     intro I B hI hImax hBmax
-    simp only [mem_maximals_iff, mem_setOf_eq, not_and, not_forall, exists_prop,
-      exists_and_left, iff_true_intro hI, true_imp_iff] at hImax hBmax
-    obtain ⟨I', hI', hII', hne⟩ := hImax
-    obtain ⟨e, heI', heI⟩ := exists_of_ssubset (hII'.ssubset_of_ne hne)
-    have hins : Indep (insert e I) := indep_subset hI' (insert_subset heI' hII')
-    obtain (heB | heB) := em (e ∈ B)
+    obtain ⟨e, heI, hins⟩ := exists_insert_of_not_maximal indep_subset hI hImax
+    by_cases heB : e ∈ B
     · exact ⟨e, ⟨heB, heI⟩, hins⟩
+
```

Comment:
```suggestion
```
(I assume those are typos? since a refactor is not really the place to orthogonally prettify proofs)